### ISSUE TITLE
Migrations Fail: `Unknown column '|' in 'field list'`

### ISSUE THREAD

**Issue by nbarrow-inspire-labs** [envkey/envkey#100]
Hello: 

We're new to the envkey community. On a brand new MySQL8 instance with an empty database, we are getting the following error when starting envkey community for the first time (I tried both the latest release, `2.3.5` and went back to `2.2.0` to try that as well, but no luck/same error): 

```
{"msg":"envkey-api binary start","api_binary_version":"2.2.0","built_from":"de5080d18d4c652facd890cc9ed7bb45624044db","bundle_type":"api.community","built_at":"2022-10-27T00:40:06Z"}
{"ts":"2023-12-22T20:08:12.735Z","msg":"EnvKey API Community Edition is starting..."}
{"ts":"2023-12-22T20:08:12.739Z","msg":"Waiting for DB..."}
{"ts":"2023-12-22T20:08:12.766Z","msg":"Checking DB migrations..."}
{"ts":"2023-12-22T20:08:13.767Z","msg":"Available migrations","migrations":["./202001030000000_upgrade_locks.ts","./202001123000000_upgrade_locks_remove.ts","./20200118173558_init.ts","./20200904000000_active_hosts.ts","./20201028000000_secondary_index.ts","./20201112000000_exclude_from_deleted_graph.ts","./202101220000000_transaction_ids.ts","./202101220000001_update_transaction_primaries.ts","./202101250000000_multi_loggable_types.ts","./202101280000000_ips.ts","./202106130000000_objects_full_key.ts","./202107140000000_org_stats.ts","./202109220000000_tertiary_index.ts","./202112120000000_self_hosted_init.ts","./202203030000000_track_active_connections.ts","./202203160000000_active_sockets.ts","./202203180000000_recreate_active_hosts_sockets.ts","./202204040000000_cloud_stats_update.ts","./202206230000000_inheritance_fix.ts","./202207080000000_cloud_billing.ts","./202209140000000_custom_license.ts","./202209210000000_init_billing_existing_orgs.ts"]}
applied indexes idx_actor idx_actor_with_ip
applied indexes idx_device idx_device_with_ip
applied indexes idx_base idx_base_with_ip
applied indexes idx_actor idx_actor_with_ip
applied indexes idx_device idx_device_with_ip
applied indexes idx_base idx_base_with_ip
applied indexes idx_actor idx_actor_with_ip
applied indexes idx_device idx_device_with_ip
applied indexes idx_base idx_base_with_ip
`transaction_ids` table created. Populating...
Batch 0. Fetching batch from 'logs'
No more log rows. Breaking.
Updating transaction_ids_by_target_id primary key. Might take awhile...
Adding additional loggableType columns to logs and rebuilding indexes. Might take awhile...
dropped indexes idx_actor idx_actor_with_ip
dropped indexes idx_device idx_device_with_ip
dropped indexes idx_base idx_base_with_ip
applied indexes idx_actor idx_actor_with_ip
applied indexes idx_device idx_device_with_ip
applied indexes idx_base idx_base_with_ip
Adding additional loggableType columns to transaction_ids and rebuilding indexes. Might take awhile...
dropped indexes idx_actor idx_actor_with_ip
dropped indexes idx_device idx_device_with_ip
dropped indexes idx_base idx_base_with_ip
applied indexes idx_actor idx_actor_with_ip
applied indexes idx_device idx_device_with_ip
applied indexes idx_base idx_base_with_ip
Adding additional loggableType columns to transaction_ids_by_target_id and rebuilding indexes. Might take awhile...
dropped indexes idx_actor idx_actor_with_ip
dropped indexes idx_device idx_device_with_ip
dropped indexes idx_base idx_base_with_ip
applied indexes idx_actor idx_actor_with_ip
applied indexes idx_device idx_device_with_ip
applied indexes idx_base idx_base_with_ip
Added fullKey column. Now adding fullKey to existing rows...
migration file "202106130000000_objects_full_key.ts" failed
migration failed with error: Unknown column '|' in 'field list'
{"ts":"2023-12-22T20:08:15.225Z","msg":"DB migrations were unsuccessful.","alert":true,"err":{"message":"Unknown column '|' in 'field list'","code":"ER_BAD_FIELD_ERROR","errno":1054,"sqlState":"42S22","sqlMessage":"Unknown column '|' in 'field list'","name":"Error","stack":"Error: Unknown column '|' in 'field list'\n    at c.query (/tmp/webpack:/Users/daneschneider/projects/envkey/v2/node_modules/mysql2/promise.js:92:22)\n    at Object.t.transactionQuery (/tmp/webpack:/Users/daneschneider/projects/envkey/v2/public/app/api/shared/src/db_fns.ts:82:41)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at Object.t.up (/tmp/webpack:/migrations/202106130000000_objects_full_key.ts:9:15)"}}
Fatal error
exit status 1
```

This error seems to be related to https://github.com/envkey/envkey/blob/main/public/app/api/runtimes/express/migrations/202106130000000_objects_full_key.ts. How best should we proceed? 

I am running using `dotenv -e /secrets-manager/.env /secrets-manager/envkey-api` with the following `.env` file
```
NODE_ENV=production
COMMUNITY_AUTH_HASH=REMOVED
SENDER_EMAIL=REMOVED
DATABASE_HOST=REMOVED
DATABASE_PORT=REMOVED
DATABASE_NAME=defaultdb
DATABASE_CREDENTIALS_JSON={"user": "REMOVED", "password": "REMOVED"}
SMTP_TRANSPORT_JSON={"host": "smtp.postmarkapp.com", "port": "2525", "auth": { "user": "REMOVED", "pass": "REMOVED" }}

```

**Comment by danenania** [comment]
This should hopefully be fixed in 2.3.6. While we haven't been able to reproduce it with our environment/MySQL configuration, it seems likely that it was caused by using double quotes around the `|` in the migration rather than single quotes. Both single and double quotes are working in our tests, but it seems that double quotes could cause MySQL to interpret the `|` as a table name, which would lead to an error like the one you got.

Closing for now.

**Comment by nbarrow-inspire-labs** [comment]
Awesome thanks @danenania I'll take a look, appreciate the quick turnaround!
